File: app/src/main/java/com/ckey/demo/adapter/WordAdapter.java
package com.ckey.demo.adapter;

import android.content.Context;
import android.text.TextUtils;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.request.RequestOptions;
import com.ckey.demo.R;
import com.dictionary.sdk.bean.ChineseCharacter;
import com.youth.banner.adapter.BannerAdapter;
import com.youth.banner.util.BannerUtils;

import java.util.List;

public class WordAdapter extends BaseAdapter {

    private Context context;
    private List<ChineseCharacter> list;

    public WordAdapter(Context contexts, List<ChineseCharacter> list) {
        this.context = contexts;
        this.list = list;
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {
        ViewHold viewHold = new ViewHold();
        if (convertView == null) {
            convertView = View.inflate(context, R.layout.layout_word, null);
            viewHold.tvpinyin = (TextView) convertView.findViewById(R.id.tvpinyin);
            viewHold.tvTitle = (TextView) convertView.findViewById(R.id.tvTitle);
            convertView.setTag(viewHold);
        } else {
            viewHold = (ViewHold) convertView.getTag();
        }

        ChineseCharacter date = list.get(position);

        viewHold.tvTitle.setText(date.characterName);
        viewHold.tvpinyin.setText(date.characterPinyin);

        return convertView;
    }

    class ViewHold {
        private TextView tvTitle,tvpinyin;
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }


}

File: app/src/main/java/com/ckey/demo/adapter/WordAdapter.kt
package com.ckey.demo.adapter

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.ckey.demo.R
import com.ckey.demo.bean.ChineseCharacterInfo
import com.dictionary.sdk.log.XLog

class WordAdapter(private val context: Context, list: List<ChineseCharacterInfo>) : BaseAdapter() {
    private val list: List<ChineseCharacterInfo>

    init {
        this.list = list
    }

    override fun getCount(): Int {
        return list.size
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        var convertView: View? = convertView
        var viewHold : ViewHold = ViewHold()
        if (convertView == null) {
            convertView = View.inflate(context, R.layout.layout_word, null)
            viewHold.tvpinyin = convertView.findViewById<View>(R.id.tvpinyin) as TextView
            viewHold.tvTitle = convertView.findViewById<View>(R.id.tvTitle) as TextView
            convertView.tag = viewHold
        } else {
            viewHold = convertView.tag as ViewHold
        }
        var data: ChineseCharacterInfo = list[position]
        viewHold.tvTitle!!.text = data.characterName
        viewHold.tvpinyin!!.text = data!!.characterPinyin
        return convertView
    }

    internal inner class ViewHold {
        var tvTitle: TextView? = null
        var tvpinyin: TextView? = null
    }

    override fun getItem(position: Int): Any? {
        return null
    }

    override fun getItemId(position: Int): Long {
        return 0
    }
}